storage/reads/response_writer.gen.go

```diff
@@ -413,39 +469,53 @@ func (w *ResponseWriter) streamBooleanArrayPoints(cur cursors.BooleanArrayCursor
 	frame := p.BooleanPoints
 	w.res.Frames = append(w.res.Frames, datatypes.ReadResponse_Frame{Data: p})
 
-	var (
-		seriesValueCount = 0
-		b                = 0
-	)
-
+	var seriesValueCount = 0
 	for {
+		// If the number of values produced by cur > 1000,
+		// cur.Next() will produce batches of values that are of
+		// length ≤ 1000.
+		// We attempt to limit the frame Timestamps / Values lengths
+		// the same to avoid allocations. These frames are recycled
+		// after flushing so that on repeated use there should be enough space
+		// to append values from a into frame without additional allocations.
```

Comment:
This makes sense. I'm not sure if I understand all of the memory implications of this change. WhatI perceive is that generally, you're limiting the possible size of each batch to reduce wasted excess allocations. Let me know if I'm missing the point here.